Update using-thread-local-storage.md

huzaifa-d opened the pull request:
Changing code to be out-of-the box compatible with C++ 14.

huzaifa-d committed:
Update using-thread-local-storage.md

Changing code to be out-of-the box compatible with C++ 14.

PRMerger20 commented:
@theShaan : Thanks for your contribution! The author(s) have been notified to review your proposed change.

stevewhims committed:
Yes, having it const is good...

However, if we use LPCSTR then we're staying in the spirit of the original and getting exactly the same type.

stevewhims commented:
Thanks for the PR, @theShaan. I've changed the macro from LPSTR (which is not const) to LPCSTR (which is const). That achieves the same good ends as your change does, but it also stays in the same spirit as the original code (which was to use the Windows type macros).
